rosetta/retriever/retriever.go

```diff
@@ -231,44 +231,36 @@ func (r *Retriever) Block(id identifier.Block) (*object.Block, []identifier.Tran
 		count++
 	}
 
+	h := header.Height - 1
```

Comment:
This could underflow, so it doesn't work this way. You have to reverse the condition and the order, like I had it in the example code. Initialize the parent to be the block itself by default, and only change it if height is bigger than zero.

Reply:
For me, it's more logical to initialize something with the hard-coded value, and then have a conditional to set it to a derived value.